packages/auth/src/client/utils/store/signInStore.ts

```diff
@@ -19,46 +21,113 @@ type SignInAction =
 	| { type: 'SET_SIGN_IN_STATE'; value: SignInState }
 	| { type: 'SET_USERNAME'; value?: string }
 	| { type: 'SET_CHALLENGE_NAME'; value?: ChallengeName }
-	| { type: 'SET_SIGN_IN_SESSION'; value?: string };
+	| { type: 'SET_SIGN_IN_SESSION'; value?: string }
+	| { type: 'RESET_STATE' };
+
+// Minutes until stored session invalidates
+const MS_TO_EXPIRY = 3 * 60 * 1000; // 3 mins
+const TGT_STATE = 'CognitoSignInState';
+const SIGN_IN_STATE_KEYS = {
+	username: `${TGT_STATE}.username`,
+	challengeName: `${TGT_STATE}.challengeName`,
+	signInSession: `${TGT_STATE}.signInSession`,
+	expiry: `${TGT_STATE}.expiry`,
+};
 
 const signInReducer: Reducer<SignInState, SignInAction> = (state, action) => {
 	switch (action.type) {
 		case 'SET_SIGN_IN_SESSION':
+			persistSignInState({ signInSession: action.value });
+
 			return {
 				...state,
 				signInSession: action.value,
 			};
+
 		case 'SET_SIGN_IN_STATE':
+			persistSignInState(action.value);
+
 			return {
 				...action.value,
 			};
+
 		case 'SET_CHALLENGE_NAME':
+			persistSignInState({ challengeName: action.value });
+
 			return {
 				...state,
 				challengeName: action.value,
 			};
+
 		case 'SET_USERNAME':
+			persistSignInState({ username: action.value });
+
 			return {
 				...state,
 				username: action.value,
 			};
+
 		case 'SET_INITIAL_STATE':
-			return defaultState();
+			return initializeState();
+
+		case 'RESET_STATE':
+			clearPersistedSignInState();
+
+			return getDefaultState();
+
+		// this state is never reachable
 		default:
 			return state;
 	}
 };
 
-function defaultState(): SignInState {
-	return {
-		username: undefined,
-		challengeName: undefined,
-		signInSession: undefined,
-	};
-}
+const isExpired = (expiryDate: string | null): boolean => {
+	const expiryTimestamp = Number(expiryDate);
+	const currentTimestamp = Date.now();
+
+	return expiryTimestamp <= currentTimestamp;
+};
+
+const clearPersistedSignInState = () => {
+	for (const storedKey of Object.values(SIGN_IN_STATE_KEYS)) {
```

Comment:
Nit: Align terminology
```suggestion
	for (const stateKey of Object.values(SIGN_IN_STATE_KEYS)) {
```